### PR TITLE
docs: update RELEASING.md with non-main branch instructions

### PR DESCRIPTION
**Pull request by sipsma**
Updates to instructions on different process when releasing from non-main branch.
* Also have some small fixes for little things I noticed as extra commits

I also created separate issues for some of the non-trivial problems I hit during the release that are mentioned in these updated docs:
* https://github.com/dagger/dagger/issues/7756
* https://github.com/dagger/dagger/issues/7757

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -303,17 +322,15 @@ go mod tidy
 cd ..
 ```
 
-- [ ] Update all dagger versions in `.github/` to `$ENGINE_VERSION`
-      - The version numbers (of the form `<major>.<minor>.<patch>`) should be updated to the new version
-      - The worker runner versions (of the form `dagger-v<major>-<minor>-<patch>-<worker>`)
+- [ ] Update all dagger versions in `.github/` to `$ENGINE_VERSION` - The version numbers (of the form `<major>.<minor>.<patch>`) should be updated to the new version - The worker runner versions (of the form `dagger-v<major>-<minor>-<patch>-<worker>`)
```
Comment (jedevc) [inline RELEASING.md:325]: nit: looks like these accidentally got joined?

Comment (sipsma) [reply RELEASING.md]: Fixed!

**Comment by sipsma** [comment]
@gerhard @jedevc Got the updated instructions in a readable-enough state; basic idea is adding collapsible sections for the special instructions when doing a release off a non-main branch so they are hard-to-miss without making it too hard to follow. Let me know what you think.

File: RELEASING.md [hunk 2]
```diff
@@ -172,33 +182,91 @@ and improve it. We want small, constant improvements which compound. Therefore:
...
+>
+> `refusing to allow an OAuth App to create or update workflow` when pushing.
+>
+> If you encounter this, you may need to change your git remote from https to ssh, e.g.
+>
+> `git remote set-url $DAGGER_REPO_REMOTE git@github.com:dagger/dagger.git`
+>
+> It's advisable to go back to https after the release is complete to not retain excessive permissions longer than necessary, e.g. to go back run
+>
+> `git remote set-url $DAGGER_REPO_REMOTE https://github.com/dagger/dagger.git`
```
Comment (jedevc) [inline RELEASING.md:212]: Hm, this one is weird. Not sure why this happens, I generally use `ssh` remotes everywhere, I've never seen this one.

cc @gerhard have we changed permissions at all recently?

**Review by jedevc — APPROVED** [review]

File: RELEASING.md (hunk 2 above)
Comment (sipsma) [reply RELEASING.md:212]: Oddly it was inconsistent when I got it. I pushed the `sdk/go/v0.11.9` tag no problem but then hit this when pushing python in the next steps?
